File: problems/starterpackages/SteinerStarter.java
import java.util.*;
import java.io.*;

public class SteinerStarter {
	// Helper classes for dealing with points and edges.
	private static class Point {
		public double x, y;
		public Point(double x, double y) {
			this.x = x;
			this.y = y;
		}
	}
	private static class Edge {
		public Point p1, p2;
		public Edge(Point p1, Point p2) {
			this.p1 = p1;
			this.p2 = p2;
		}
	}
	
	// Gets a problem from a file as an ArrayList of points.
	public static ArrayList<Point> getProblem(String filename) {
		File fileEntry = new File(filename);
		ArrayList<Point> points = new ArrayList<Point>(0);
		try (BufferedReader br = new BufferedReader(new FileReader(fileEntry))) {
			String line;
			while ((line = br.readLine()) != null) {
				String[] components = line.split(" ");
				points.add(new Point(Double.parseDouble(components[0]), Double.parseDouble(components[1])));
			}
		} catch(Exception e) {
			e.printStackTrace();
			System.out.println("There was an error reading in the input problem.");
		}
		return points;
	}
	
	// Outputs an ArrayList of edges to file out.txt for submission.
	public static void outputSolutionsToFile(ArrayList<Edge> solution) {
		String filename = "out.txt", content = new String();
		for(Edge i : solution) content += i.p1.x + " " + i.p1.y + " " + i.p2.x + " " + i.p2.y + "\n";
		try {
			FileWriter writer = new FileWriter(new File(filename), false);
            writer.write(content);
            writer.close();
		} catch(Exception e) {
			e.printStackTrace();
			System.out.println("There was an error outputting your solution.");
		}
	}
	
	public static void main(String [] args) {
		ArrayList<Point> problem = getProblem("st.txt"); // Gets the problem from st.txt as an ArrayList of points.
		
		ArrayList<Edge> solution = new ArrayList<Edge>(); // Where you should put the edges that make up your solution tree.
		
		// This code creates edges in the order the points were given.
		for(int a = 1; a < problem.size(); a++) {
			solution.add(new Edge(problem.get(a-1), problem.get(a)));
		}

		outputSolutionsToFile(solution); // Outputs your solution to out.txt for submission.
	}
}

File: problems/starterpackages/SteinerStarter.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <stdexcept>
#include <vector>

// Helper structs for dealing with points and edges.
struct Point {
	double x, y;
};
struct Edge {
	Point p1, p2;
};

// Gets a problem from a file as a vector of points.
std::vector<Point> getProblem(const std::string & filename) {
	std::vector<Point> pts(1000); //Number of points
	std::ifstream input(filename);
	if(input.is_open()) for(auto a = pts.begin(); a != pts.end(); a++) input >> a->x >> a->y;
	else {
		std::cout << "Couldn't open input file.";
		getchar();
		exit(EXIT_FAILURE);
	}
	input.close();
	return pts;
}

// Outputs a vector of edges to file out.txt for submission.
void outputSolutionsToFile(const std::vector<Edge> & s) {
	std::ofstream output("out.txt");
	if(output.is_open()) for(auto a = s.begin(); a != s.end(); a++) output << a->p1.x << ' ' << a->p1.y << ' ' << a->p2.x << ' ' << a->p2.y << '\n';
	else {
		std::cout << "Couldn't open output file";
		getchar();
		exit(EXIT_FAILURE);
	}
	output.close();
}

int main() {

	std::vector<Point> problem;
	problem = getProblem("st.txt"); // Gets the problem from st.txt as a vector of points.

	std::vector<Edge> solution; // Where you should put the edges that make up your solution tree.

	// This code creates edges in the order the points were given.
	for(auto a = problem.begin(); a != problem.end(); a++) {
		Edge e{ *a, *a };
		solution.push_back(e);
	}

	outputSolutionsToFile(solution); // Outputs your solution to out.txt for submission.

	return EXIT_SUCCESS;
}